tests/unit/amo/components/TestAddon.js

```diff
@@ -761,23 +763,26 @@ describe(__filename, () => {
     expect(button.prop('disabled')).toEqual(false);
   });
 
-  it('disables install switch for unsupported clients', () => {
+  it('disables install btn for incompatibility with firefox version', () => {
     const root = shallowRender({
-      getClientCompatibility: getClientCompatibilityFalse,
+      getClientCompatibility: () => ({
+        compatible: false,
+        downloadUrl: 'https://www.seamonkey-project.org',
+        reason: INCOMPATIBLE_UNDER_MIN_VERSION,
+      }),
     });
     expect(root.find(InstallButton).prop('disabled')).toBe(true);
   });
 
-  it('passes the downloadUrl from getClientCompatibility', () => {
+  it('hides banner on non firefox clients and displays firefox download btn', () => {
     const root = shallowRender({
       getClientCompatibility: () => ({
         compatible: false,
-        downloadUrl: 'https://www.seamonkey-project.org',
         reason: INCOMPATIBLE_NOT_FIREFOX,
```

Comment:
Maybe you could keep this test almost unchanged by only modifying the `reason` here:

```diff
diff --git a/tests/unit/amo/components/TestAddon.js b/tests/unit/amo/components/TestAddon.js
index cc9cbd1..6969179 100644
--- a/tests/unit/amo/components/TestAddon.js
+++ b/tests/unit/amo/components/TestAddon.js
@@ -763,26 +763,27 @@ describe(__filename, () => {
     expect(button.prop('disabled')).toEqual(false);
   });
 
-  it('hides banner on non firefox clients and displays firefox download btn', () => {
+  it('passes the downloadUrl from getClientCompatibility', () => {
     const root = shallowRender({
       getClientCompatibility: () => ({
         compatible: false,
-        reason: INCOMPATIBLE_NOT_FIREFOX,
+        downloadUrl: 'https://www.seamonkey-project.org',
+        reason: INCOMPATIBLE_UNDER_MIN_VERSION,
       }),
     });
-    expect(root.find(AddonCompatibilityError)).toHaveLength(0);
-    expect(root.find(Button)).toHaveLength(1);
+    expect(root.find(AddonCompatibilityError).prop('downloadUrl'))
+      .toEqual('https://www.seamonkey-project.org');
   });
 
   it('passes installStatus to installButton, not add-on status', () => {

```

Reply:
@willdurand  - i see what you're saying but should I also add a test that cks for non FF browsers - in other words it checks that they show this ~"get FF now" button and shows in this case that a banner will not be displayed...  
does that make sense?

Reply:
yes we also need a test for what you've added.